trip_app/core/utils.py

```diff
@@ -19,18 +23,40 @@ def raise_for_status(resp):
         raise Exception(http_error_msg)
 
 
-def str_to_geopoint(data):
+def str_to_geopoint_or_validation_error(data):
     """ Convert string coordinates into a  Point object.
         Regex is used to handle different incoming formats.
         E.g. '23.4 23.5', '[23.4 23.5]', '(23.4 23.5)' would be converted into
-        the same object Point(23.4, 23.5) """
-    lon, lat = re.findall(r"[-+]?\d*\.?\d+|\d+", data)
-    if not -90 <= float(lon) <= 90:
+        the same object Point(23.4, 23.5). """
+    try:
+        lat_str, lon_str = re.findall(r"[-+]?\d*\.?\d+|\d+", data)
+        lat, lon = float(lat_str), float(lon_str)
```

Comment:
I think its better to make like this:
```
latlon = re.findall(r"[-+]?\d*\.?\d+|\d+", data)
if not latlon:
    raise ValidationError(f"Cannot parse {data} into geo coordinates")
validate_geo_point(*latlon)
return Point(*latlon, srid=4326)
```

Reply:
Yep that make sense as well, the reason why I'd preferred catching exception is because converting to float potentially can raise ValidationError as well, though regex should return either empty array or valid numbers, however I want to ensure more in this.

Reply:
Casting to float can not throw an error because re.findall would return you array of digits in string representation or empty array.

Reply:
Your statement is correct until this re is correct ;)

Reply:
@Meathookseed Updated to your version. Thanks.